### ISSUE TITLE
"window.restoreWindows": "none" incorrect

### ISSUE THREAD

**Issue by q00u** [microsoft/vscode#92599]
Issue Type: <b>Bug</b>

**Restore Windows**

> Never reopen a window. Always start with an empty one.

This is incorrect.

I had a hot session with unsaved changes in the past. It required other people, so I haven't gone back to that project in a long time.

After the latest update to 1.43.0, that session now launches whenever I run VS Code.

The `window.restoreWindows` setting is `none`, but instead of restoring none windows, it's restoring this old session.

But, because some of the files are still being edited (which I can't complete without other people (who are busy)), I don't want to work on this old project now. I'd like it to have the previous (seemingly intended) behavior of NOT opening any projects.

**Hot Exit**

Now, if `"files.hotExit": "onExitAndWindowClose"` instead of the default `onExit`, it goes back to the previous behavior. If I launch VS Code, even though a hot session _exists_, it doesn't restore it on launch.

So it seems that the capability to restore NONE windows sure exists. So why isn't restoreWindows=none always respected?

tl;dr
`restoreWindows=none` is only respected if a different setting is changed from default.

VS Code version: Code 1.43.0 (78a4c91400152c0f27ba4d363eb56d2835f9903a, 2020-03-09T19:47:57.235Z)
OS version: Windows_NT x64 10.0.18362

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4600M CPU @ 2.90GHz (4 x 2893)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.89GB (3.69GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

**Comment by gjsjohnmurray** [comment]
Related to #92467
